sycl/doc/design/CommandGraph.md

```diff
@@ -203,3 +203,42 @@ Level Zero:
 
 Future work will include exploring L0 API extensions to improve the mapping of
 UR command-buffer to L0 command-list.
+
+### OpenCL - cl_khr_command_buffer
+
+There are some gaps in both the OpenCL and UR specifications for Command Buffers shown in the list below. There are implementations in the UR OpenCL adapter where there is matching support for each function in the list.
+
+| UR | OpenCL | Supported |
+| --- | --- | --- |
+| urCommandBufferCreateExp | clCreateCommandBufferKHR | Yes |
+| urCommandBufferRetainExp | clRetainCommandBufferKHR | Yes |
+| urCommandBufferReleaseExp | clReleaseCommandBufferKHR | Yes |
+| urCommandBufferFinalizeExp | clFinalizeCommandBufferKHR | Yes |
+| urCommandBufferAppendKernelLaunchExp | clCommandNDRangeKernelKHR | Yes |
+| urCommandBufferAppendUSMMemcpyExp |  | No |
+| urCommandBufferAppendUSMFillExp |  | No |
+| urCommandBufferAppendMembufferCopyExp | clCommandCopyBufferKHR | Yes |
+| urCommandBufferAppendMemBufferWriteExp |  | No |
+| urCommandBufferAppendMemBufferReadExp |  | No |
+| urCommandBufferAppendMembufferCopyRectExp | clCommandCopyBufferRectKHR | Yes |
+| urCommandBufferAppendMemBufferWriteRectExp |  | No |
+| urCommandBufferAppendMemBufferReadRectExp |  | No |
+| urCommandBufferAppendMemBufferFillExp | clCommandFillBufferKHR | Yes |
+| urCommandBufferEnqueueExp | clEnqueueCommandBufferKHR | Yes |
+|  | clCommandBarrierWithWaitListKHR | No |
+|  | clCommandCopyImageKHR | No |
+|  | clCommandCopyImageToBufferKHR | No |
+|  | clCommandFillImageKHR | No |
+|  | clGetCommandBufferInfoKHR | No |
+
+Many of the OpenCL functions take a `cl_command_queue` parameter which is not present in most of the UR functions. Instead when a new command buffer is created in `urCommandBufferCreateExp` we also create and maintain a new internal `ur_queue_handle_t` with a reference stored inside of the `ur_exp_command_buffer_handle_t_` struct. This internal queue is then used with the various Append functions. The internal queue is retained and released whenever the owning command buffer is retained or released.
+
```

Comment:
Can this section document what the gaps in SYCL support are. e.g Can't use memcpy operations in a graph node. This should reflect the features used in E2E test we haven't enabled for OpenCL.

Reply:
To elaborate on this point, because I think one of the most important parts of this PR is that we effectively communicate what's supported and what's not.

* Can you make this limitation information I've asked for a new sub-section in the document, because we'll probably need to link it to people.
* Could you also document how the gaps fail, because I'd like to make sure that exceptions or asserts are thrown (i'd guess in finalize()) when unsupported functionality is used. We want to to give a good user experience, rather than the segfaulting or the end result being subtly wrong.